jupyter-extension/jupyterlab_pachyderm/pfs_manager.py

```diff
@@ -469,12 +474,62 @@ def current_datum(self) -> dict:
             all_datums_received=True,
         )
 
-    def _get_fileinfo_path(self, fileinfo: pfs.FileInfo) -> Path:
+    def download(self):
+        if len(self._datum_list) == 0 or len(self._datum_list[self._datum_index].data) == 0:
+            raise ValueError("Attempting to download empty or unmounted datum")
+
+        download_dir = (
+            f"{self._DOWNLOAD_DIR}/datum-{datetime.datetime.now().isoformat()}"
+        )
+        try:
+            os.makedirs(download_dir, exist_ok=True)
+            for fileinfo in self._datum_list[self._datum_index].data:
+                path = self._get_download_path(
+                    download_dir=download_dir, fileinfo=fileinfo
+                )
+                os.makedirs(path.parent, exist_ok=True)
+                if fileinfo.file_type == pfs.FileType.FILE:
+                    # download individual file
+                    data = list(self._client.pfs.get_file(file=fileinfo.file))[0].value
```

Comment:
Definitely don't do this, this will result in data loss. You should do:
```python
with self._client.pfs.pfs_file(file=fileinfo.file) as datum_file:
    ...
```

Reply:
good catch, thank you!